lightgbm/src/test/python/synapseml/lightgbm/test_lightgbm.py

```diff
@@ -0,0 +1,122 @@
+import os
+import shutil
+import json
+import unittest
+from pyspark.sql import SparkSession, Row
+from pyspark.ml import PipelineModel, Transformer
+from pyspark.ml.util import DefaultParamsReadable, DefaultParamsWritable
+from pyspark.ml.linalg import Vectors, VectorUDT
+from pyspark.sql.types import ArrayType, IntegerType
+import pyspark.sql.functions as f
+from synapse.ml.stages import SelectColumns
+import synapse.ml.lightgbm as lgbm
+
+class StringArrayToVectorTransformer(Transformer, DefaultParamsReadable, DefaultParamsWritable):
+    """
+    A pure-Python Transformer that converts a string representation of an integer array
+    into a Dense Vector. For example, it converts "[1, 2, 3]" into a DenseVector.
+    """
+    def __init__(self, inputCol="embedded_object_keys_string", outputCol="features", maximumVectorLength=3):
+        super(StringArrayToVectorTransformer, self).__init__()
+        self.inputCol = inputCol
+        self.outputCol = outputCol
+        self.maximumVectorLength = maximumVectorLength
+
+    def _transform(self, dataset):
+        def parse_string_embedding(s):
+            try:
+                # Convert a string like "[1, 2, 3]" to a list of integers.
+                return json.loads(s)
+            except Exception:
+                return []
+        parse_udf = f.udf(parse_string_embedding, ArrayType(IntegerType()))
+        
+        def to_vector(arr):
+            # Pad or truncate the list to ensure it has exactly `maximumVectorLength` elements.
+            if len(arr) < self.maximumVectorLength:
+                arr = arr + [0] * (self.maximumVectorLength - len(arr))
+            return Vectors.dense(arr[:self.maximumVectorLength])
+        vector_udf = f.udf(to_vector, VectorUDT())
+        
+        df = dataset.withColumn(self.outputCol, parse_udf(f.col(self.inputCol)))
+        return df.withColumn(self.outputCol, vector_udf(f.col(self.outputCol)))
+
+class LightGBMSerializationTests(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.spark = SparkSession.builder \
+            .appName("LightGBMSerializationTests") \
+            .getOrCreate()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.spark.stop()
+
+    def setUp(self):
+        # Define a temporary directory for saving models/pipelines.
+        self.temp_dir = "./tmp/lightgbm_serialization_test"
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def tearDown(self):
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
```

Comment:
Our tests manage the spark and spark context objects on behalf of the user, im not sure youll need this if you copy the structure of other tests in the repo

Reply:
Ah, I see, reused their components. Removed handling functions.